pkg/client/context.go

```diff
@@ -0,0 +1,61 @@
+package client
+
+import (
+	"context"
+	"net/http"
+)
+
+type contextKey int
+
+const (
+	authKey contextKey = iota
+	williamKey
+)
+
+func Auth(r *http.Request) *AuthInfo {
```

Comment:
Seems in these functions that we are only accessing the `context` part of the request, is that correct? If so, WDYT about directly receiving the `ctx Context` struct in the methods?

Further reading: https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39

Reply:
I leave like this in favor of how gorrila mux works. https://github.com/gorilla/mux/blob/master/mux.go#L428

Reply:
+1 for context.Context instead of http.Request - I may not even *have* an http.Request and still want to use this.